dbms/src/Storages/DeltaMerge/ReadThread/CPU.cpp

```diff
@@ -0,0 +1,69 @@
+#include <Poco/DirectoryIterator.h>
+#include <Poco/File.h>
+#include <Storages/DeltaMerge/ReadThread/CPU.h>
+
+#include <string>
+
+namespace DB::DM
+{
+// In Linux a numa node is represented by a device directory, such as '/sys/devices/system/node/node0', ''/sys/devices/system/node/node01'.
+static inline bool isNodeDir(const std::string & name)
+{
+    return name.substr(0, 4) == "node" && std::all_of(name.begin() + 4, name.end(), [](unsigned char c) { return std::isdigit(c); });
```

Comment:
do we need to ensure name.length() > 4?

Reply:
I think this needs better protection. How about using regex?

Reply:
I think check `name.size() > 4` is simple and enought.
 
regex can slove this problem but I think it is not necessarily.